plugin/azure/azure.go

```diff
@@ -138,43 +138,43 @@ func (h *Azure) updateZones(ctx context.Context) error {
 
 func updateZoneFromPublicResourceSet(recordSet publicdns.RecordSetListResultPage, newZ *file.Zone) {
 	for _, result := range *(recordSet.Response().Value) {
-		resultFqdn := *(result.RecordSetProperties.Fqdn)
-		resultTTL := uint32(*(result.RecordSetProperties.TTL))
-		if result.RecordSetProperties.ARecords != nil {
-			for _, A := range *(result.RecordSetProperties.ARecords) {
+		resultFqdn := *(result.Fqdn)
```

Comment:
Interesting, I'm wondering what the linter is thinking here.

Maybe we should make some of these changes separately?

Reply:
It looks OK to me. RecordSetProperties is embedded in the result struct, and Fqdn is unambiguous.

Reply:
This is coming from staticcheck https://staticcheck.dev/docs/checks#QF1008 
https://brunoscheufler.com/blog/2022-01-23-important-rules-of-embedding-types-in-go